java/semantickernel-api/src/main/java/com/microsoft/semantickernel/semanticfunctions/KernelFunctionFromMethod.java

```diff
@@ -167,89 +167,102 @@ private static MethodDetails getMethodDetails(
-                                return (T) ((Iterable<?>) it).iterator().next();
-                            } else {
-                                return (T) it;
-                            }
-                        });
-
-                    return r
-                        .map(it -> {
-                            // If given a variable type, use it. 
-                            // If it's wrong, then it's a programming error on the part of the caller.
-                            if (variableType != null) {
-                                return new FunctionResult<>(new ContextVariable<>(variableType, it));
-                            }
-
-                            Class<?> returnParameterType = function
-                                .getMetadata()
-                                .getReturnParameter()
-                                .getParameterType();
-
-                            // If the function has a return type that has a ContextVariableType<T>, use it.
-                            ContextVariableType<T> contextVariableType = getContextVariableType(returnParameterType);
-                            if (contextVariableType == null) {
-                                // If getting the context variable type from the function fails, default to
-                                // using the NoopConverter.
-                                contextVariableType = getDefaultContextVariableType(returnParameterType);
-                            }
+            try {
+                List<Object> args =
+                    Arrays.stream(method.getParameters())
+                        .map(getParameters(method, updatedArguments, kernel))
+                        .collect(Collectors.toList());
+
+                Mono<?> mono;
+                if (method.getReturnType().isAssignableFrom(Mono.class)) {
+                    mono = (Mono<?>) method.invoke(instance, args.toArray());
+                } else {
+                    mono = invokeAsyncFunction(method, instance, args);
+                }
 
-                            if (contextVariableType != null) {
-                                return new FunctionResult<>(new ContextVariable<>(contextVariableType, it));
+                Mono<T> r = mono
+                    .map(it -> {
+                        if (it instanceof Iterable) {
+                            // Handle return from things like Mono<List<?>>
+                            // from {{function 'input'}} as part of the prompt.
+                            return (T) ((Iterable<?>) it).iterator().next();
+                        } else {
+                            return (T) it;
+                        }
+                    });
+
+                return r
+                    .map(it -> {
+                        // If given a variable type, use it.
+                        // If it's wrong, then it's a programming error on the part of the caller.
+                        if (variableType != null) {
```

Comment:
For void functions, does the code even get to here if null is returned? If we map void return to Mono.empty(), then we for sure get here, but you end up returning a FunctionResult<Mono> which isn't the right thing. At least that's what the code did before these changes. So I guess the question is, what gets returned when the void methods are called in Example09?

Reply:
this code is not reached for a void function, you end up returning a function of `Mono<FunctionResult>`, which imo is correct as you end up with a mono that never returns a result

Reply:
If I have a chain of function calls in the reactive stream and one of the functions is a void function, the stream is broken.